src/ucp/core/ucp_mm.c

```diff
@@ -706,40 +742,64 @@ ucs_status_t ucp_mem_map(ucp_context_h context, const ucp_mem_map_params_t *para
             status = UCS_ERR_INVALID_PARAM;
             goto out;
         }
-
         mem_type = params->memory_type;
     }
 
+    if (exported_memh_buffer != NULL) {
+        memh_flags |= UCP_MEM_FLAG_IMPORTED;
+
+        if (*(uint16_t*)exported_memh_buffer == sizeof(ucp_memh_dummy_buffer)) {
+            goto out_zero_mem;
+        }
+    } else if (length == 0) {
+        goto out_zero_mem;
+    }
+
+    if (address == NULL) {
+        if (!(flags & UCP_MEM_MAP_ALLOCATE) && (length > 0)) {
+            ucs_error("undefined address with nonzero length requires "
+                      "UCP_MEM_MAP_ALLOCATE flag");
+            status = UCS_ERR_INVALID_PARAM;
+            goto out;
+        }
+    } else if ((!(flags & UCP_MEM_MAP_ALLOCATE) &&
+                (flags & UCP_MEM_MAP_FIXED))) {
+        ucs_error("wrong combination of flags when address is defined");
+        status = UCS_ERR_INVALID_PARAM;
+        goto out;
+    }
+
     uct_flags = ucp_mem_map_params2uct_flags(params);
 
-    if (flags & UCP_MEM_MAP_ALLOCATE) {
+    if (memh_flags & UCP_MEM_FLAG_IMPORTED) {
+        status = ucp_memh_import(context, exported_memh_buffer, &memh);
+    } else if (flags & UCP_MEM_MAP_ALLOCATE) {
```

Comment:
pls print error and return for unsupported flags combination (e.g. ALLOCATE | IMPORTED)